Some other (important!) wxQt fixes

AliKet opened the pull request:
Please review and test before merging. TIA!

Commit:
Remove unreferenced defines from wxQt code base

Commit:
Add missing override to wxQtEventSignalHandler

Commit:
Stop using QScrollArea as a central widget of a wxFrame under wxQt

There's really no need to make the wxFrame itself scrollable as it unnecessarily
complicates the implementation and introduces annoying bugs like content shifted
by 1px. Moreover, the scrolling logic is already encapsulated in other widgets
like wxScrolledWindow which can be used as a hosting container in the frame to
handle scrolling.

Commit:
Fix handling paint events under wxQt after the last commit

Qt doesn't allow painting on a widget from a handler of another widget

Commit:
Get rid of unnecessary else statement in QtHandlePaintEvent

Commit:
Fix wxPaintDC under wxQt after the last commits

The {menu,tool,status} bars will not be painted correctly otherwise.

Commit:
Fix wxScrollWindow not scrolling by mouse dragging under wxQt

Get rid of wxQtInternalScrollBar and use QScrollArea's scrollbars instead.

Commit:
Always use live resize for wxAUI under wxQt

Commit:
Fix compilation error in auidemo sample under wxQt

Also fix ID_NotebookNoCloseButton menu item ui update which causes
OnUpdateUI() handler to run forever under wxQt.

vadz commented:
Thanks, this looks good to me, but I didn't test it.

I'll leave it here for a few days to give time to the others to check it and report any problems (or lack of them!).

Thanks again!

Commit:
Override {Set,Is}DoubleBuffered() under wxQt

Closes #24136.

Commit:
Change the return type of QtGetScrollBarsContainer() under wxQt

Neither QListWidget nor QTextEdit derive from QScrollArea and the cast to it is
just UB. Instead, using the common base class QAbstractScrollArea is safer and
eliminates the need to these dangerous and insidious crosscast'ings.

Commit:
Get rid of QtGetScrollBarsContainer() from wxWindow API under wxQt

Because It is simpler to initialize (the inherited) m_qtContainer member
directly in derived classes during construction process, and make the API
a little bit cleaner.

Commit:
Honor wxALWAYS_SHOW_SB flag if specified under wxQt

Commit:
Refactor the code that chooses which widget to use for drawing under wxQt

No real changes.

Commit:
Keep forward declarations of Qt classes in alphabetical order

Commit:
Update outdated comments under wxQt

There is no longer a QtGetContainer() function name. wxQtEventForwarder was renamed
to wxQtEventSignalHandler. And QtGetScrollBarContainer() was just removed in previous
commits.

Commit:
Get rid of QtCreateControl() from wxQt code base

If possible, all windows should call wxWindow::Create() on creation now,
because among other things:
- It manages generic and non-generic windows uniformly and transparently
- It ensures an appropriate initial size for the window
- AddChild() and PostCreation() are automatically called
- Scrollbar policies are set in one place.

vadz commented:
Should this be already merged?

vadz reviewed (approved):
Thanks, this looks like a nice simplification!

I'd just like to ask: there is this comment in wx/qt/window.h:

```cpp
    // Used by all window classes in the widget creation process.
    void PostCreation( bool generic = true );
```

but this doesn't seem to be true (any more), could this comment please be updated to explain when exactly this should be called?

AliKet commented:
> Should this be already merged?

AFAICS, people are more inclined to test master than pull requests. So yes this should be merged (after fixing review comments of course) to get some feedback from interested users if they are still around!

TIA.

AliKet commented on samples/dialogs/dialogs.cpp:
```diff
@@ -758,11 +758,13 @@ MyFrame::MyFrame(const wxString& title)
     sizer->Add(m_infoBarAdvanced, wxSizerFlags().Expand());
     SetSizer(sizer);
 
+#ifdef __WXMSW__
```
So far I'm testing under Linux Mint, Ubuntu (default theme) and KDE.
KDE is the only one affected by this:

Before:

![dialog_not_ok](https://github.com/wxWidgets/wxWidgets/assets/7704771/2262a1f2-3759-4679-aff1-206a9791ac88)

After:
![dialog_ok](https://github.com/wxWidgets/wxWidgets/assets/7704771/e2f14d80-6f97-4f8a-9c0c-c7a26d4383da)

vadz replied:
Looks like the colour is being set for the entire window and not just the main widget?

AliKet replied:
Yes, but what I don't understand is why only KDE ?
here is a screenshot of the sample under Xubuntu:

![dialog_ok](https://github.com/wxWidgets/wxWidgets/assets/7704771/00688d3f-8c5e-4859-a446-4adf920ab85e)

As you can see, the menubar and statusbar are not affected by `SetOwnBackgroundColour()`

vadz replied:
I don't know this neither but I still think that if SetBackgroundCopour() changes the menu bar colour, it's something that needs to be fixed in wxQt and not in the sample.

AliKet replied:
> Looks like the colour is being set for the entire window and not just the main widget?

According to the [Qt docs](https://doc.qt.io/qt-6/application-windows.html#window-geometry) , the **menubar**, **toolbar** and **statusbar** are considered part of the client area. 

So when the background colour of the `wxFrame` is changed via `SetBackgroundColour()`, https://github.com/wxWidgets/wxWidgets/blob/99434fa005c790a10b43947d5b286a44af90847c/src/qt/window.cpp#L1263-L1272
the above controls are also affected by this call because `GetHandle()` returns `QMainWindow`. IOW, to make `SetBackgroundColour()` works like the other ports, we should use `QtGetParentWidget()` which is the same as `GetHandle()` for any window but a `wxFrame`. in which case it returns the central widget.

I tested these changes (`SetForegroundColour()` also updated) and the issue is gone.

vadz replied:
Great, thanks, this is exactly what I hoped was the case (sorry if I hadn't explained it clearly...) and it's great to have it fixed.

We should probably restrict other functions like this to just the central widget, if this is not already the case.

But, for now, can we remove this change from the sample and merge this already?

Thanks again!

Commit:
Fix Set{Back,Fore}groundColour() for wxFrame under wxQt

Restrict the effect of these functions to the central widget only.
Otherwise the menubar, toolbar and statusbar would also be affected.

Commit:
Fix wxStaticBitmap not using correct bitmap size under wxQt

Commit:
Handle wxTextCtrl created with wxTE_DONTWRAP or wxTE_NO_VSCROLL under wxQt

AliKet commented:
`wxInfoBar` now works correctly (it didn't work before):

![Screenshot_2023-12-26_18-29-37](https://github.com/wxWidgets/wxWidgets/assets/7704771/0fe336ff-be5f-46ad-ae94-460e2d704854)

`wxStaticBitmap`  in a dialog had always the size of 16x16 (corrected now):

![Screenshot_2023-12-26_18-30-29](https://github.com/wxWidgets/wxWidgets/assets/7704771/a790704a-cbe2-40d1-832b-610657940403)

vadz commented:
Thanks, I'll merge this now as there are already more than enough things here, but please keep making new wxQt PRs if you have more things to fix.

Thanks again!
